x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java

```diff
@@ -275,21 +205,23 @@ public void getCheckpointingInfo(
         long lastCheckpointNumber,
         TransformIndexerPosition nextCheckpointPosition,
         TransformProgress nextCheckpointProgress,
-        ActionListener<TransformCheckpointingInfo> listener
+        ActionListener<TransformCheckpointingInfoBuilder> listener
     ) {
 
-        TransformCheckpointingInfoBuilder checkpointingInfoBuilder = new TransformCheckpointingInfoBuilder();
+        TransformCheckpointingInfo.TransformCheckpointingInfoBuilder checkpointingInfoBuilder =
+            new TransformCheckpointingInfo.TransformCheckpointingInfoBuilder();
 
         checkpointingInfoBuilder.setNextCheckpointPosition(nextCheckpointPosition).setNextCheckpointProgress(nextCheckpointProgress);
 
         long timestamp = System.currentTimeMillis();
+        AtomicReference<TransformCheckpoint> lastCheckpoint = new AtomicReference<>(TransformCheckpoint.EMPTY);
```

Comment:
Why do we need an atomic reference here? Seems like we could store it in `checkpointingInfoBuilder` and add a getter for the `TransformCheckpoint.getBehind` calculation so it has access to the last checkpoint

Reply:
yes, had the same thought but did not like to create a getter in a builder and therefore have some strange dependency.

Reply:
after a 2nd thought, i probably made it to complicated, so I added some setters.